### DISCUSSION TITLE
Question about fixture_batch_size in conftest.py

### DISCUSSION THREAD

**Discussion by Waschenbacher** [emdgroup/baybe#269] [Q&A] [answered]
No much experience on pytest yet, but I'm just curious about the `pytest.mark.slow` marker in `fixture_batch_size`. Should it be associated with "3", or "1"?

**Comment by Scienfitz** [comment, marked as answer]
Hey @Waschenbacher 

This is a custom marker form us. When you run pytest you can use the optional argument `--fast`. This then will skip all tests and fixture variants that have been marked with `mark.slow`. The logic implementing that is in conftest.py

The naming here is prob a bit counterintuitive as a batch recommendation with 3 is likely slower than with 1. But the logic of marking the variant with `batch_size=1` as `slow` stems from us not wanting to test both batch sizes 1 and 3 if the user wants to do `--fast`, so we decided in that case just 3 will be done as `batch_size > 1` is the more complex use case that should be covered.

We don't use the `--fast` mechanism much anyway tbh as we mostly do testing via tox

**Reply by AdrianSosic** [reply]
Yup, exactly. Plus: I think the different batch sizes will be removed soon, anyway, once our testing codebase has been cleaned up (and much of the stuff has been outsourced to hypothesis)
